src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/adjourncase/AdjournCasePreviewServiceTest.java

```diff
@@ -1794,152 +1797,112 @@ public void givenInterpreterLanguageRequiredAndProvided_thenErrorShouldNotBeInte
-        sscsCaseData.setAdjournCaseInterpreterLanguage(new DynamicList(item, Arrays.asList()));
-        final PreSubmitCallbackResponse<SscsCaseData> response = service.preview(callback, DocumentType.DRAFT_ADJOURNMENT_NOTICE, USER_AUTHORISATION, true);
-        NoticeIssuedTemplateBody templateBody = verifyTemplateBody(NoticeIssuedTemplateBody.ENGLISH_IMAGE, "Appellant Lastname", "face to face hearing", true);
+        NoticeIssuedTemplateBody templateBody = setAdjournmentAndGetTemplateBody(sscsCaseData, item);
         assertEquals("a sign language interpreter (Sign (American Sign))", templateBody.getAdjournCaseTemplateBody().getInterpreterDescription());
     }
 
     @Test
     public void givenAValidNonSignLanguageKey_ThenExtractTheInterpreterDescription(){
-        commonSetupForInterpreterDescription(sscsCaseData);
+        when(signLanguagesService.getLanguageByHmcReference(anyString())).thenReturn(null);
         DynamicListItem item = new DynamicListItem("senegalFrench", "Senegal (French) Olof Dialect");
-        sscsCaseData.setAdjournCaseInterpreterLanguage(new DynamicList(item, Arrays.asList()));
-        final PreSubmitCallbackResponse<SscsCaseData> response = service.preview(callback, DocumentType.DRAFT_ADJOURNMENT_NOTICE, USER_AUTHORISATION, true);
-        NoticeIssuedTemplateBody templateBody = verifyTemplateBody(NoticeIssuedTemplateBody.ENGLISH_IMAGE, "Appellant Lastname", "face to face hearing", true);
-        assertEquals("a sign language interpreter (Senegal (French) Olof Dialect)", templateBody.getAdjournCaseTemplateBody().getInterpreterDescription());
+        NoticeIssuedTemplateBody templateBody = setAdjournmentAndGetTemplateBody(sscsCaseData, item);
+        assertEquals("an interpreter in Senegal (French) Olof Dialect", templateBody.getAdjournCaseTemplateBody().getInterpreterDescription());
     }
 
     @Test
-    public void givenAnInvalidLanguageKey_ThenReturnNullInterpreterDescription(){
-        commonSetupForInterpreterDescription(sscsCaseData);
+    public void givenAnInvalidLanguageKey_ThenReturnErrorLanguageNotSet(){
         DynamicListItem item = new DynamicListItem("something", null);
+        sscsCaseData.setAdjournCaseGenerateNotice("yes");
+        sscsCaseData.setAdjournCaseInterpreterRequired("yes");
         sscsCaseData.setAdjournCaseInterpreterLanguage(new DynamicList(item, Arrays.asList()));
         final PreSubmitCallbackResponse<SscsCaseData> response = service.preview(callback, DocumentType.DRAFT_ADJOURNMENT_NOTICE, USER_AUTHORISATION, true);
-        NoticeIssuedTemplateBody templateBody = verifyTemplateBody(NoticeIssuedTemplateBody.ENGLISH_IMAGE, "Appellant Lastname", "face to face hearing", true);
-        assertEquals("a sign language interpreter (null)", templateBody.getAdjournCaseTemplateBody().getInterpreterDescription());
+        String error = response.getErrors().stream().findFirst().orElse("");
+        assertEquals("Language label is not set", error);
     }
```

Comment:
You dont need all these different tests, when unit testing we want to test the different routes, there is no point testing each of the different types of sign language as we are controlling what sign language is returning.
A good way to think about this is a tree where it branches the logic for every if statement or conditional